Microsoft.DotNet.Interactive.PowerShell/DollarProfileHelper.cs

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+
+namespace Microsoft.DotNet.Interactive.PowerShell
+{
+    using System.Management.Automation;
+    using System.Reflection;
+
+    internal static class DollarProfileHelper
+    {
+        private const string _profileName = "Microsoft.dotnet-interactive_profile.ps1";
+
+        // This is the easiest way to get the config directory (where the PROFILE lives), unfortunately.
+        private static readonly string _configPath = typeof(Platform).GetField(
+                "ConfigDirectory",
+                BindingFlags.Static | BindingFlags.NonPublic)
+            .GetValue(null) as string;
+
+        internal static readonly string AllUsersCurrentHost = GetFullProfileFilePath(forCurrentUser: false);
```

Comment:
these two could be static properties with getter only and single evaluation
Something like 
```
private static string _allUsersCurrentHost;
internal static string AllUsersCurrentHost { get; } = _allUsersCurrentHost?=GetFullProfilePAth(forCurrentUser: false);
```

Reply:
Auto properties have the same effect, yeah?

```csharp
internal static string AllUsersCurrentHost { get; } = GetFullProfileFilePath(forCurrentUser: false);
```

Reply:
The code you just wrote initialises the static variable with the result of the invocation, using lambdas you map the better to the function call (os is evaluated everyday time) with my snippet you have a lazy evaluation done in the first use of the property. What is the desired behaviour?

Reply:
It doesn't really matter since these props will be accessed on startup anyway.

Reply:
Because of the extension method:

https://github.com/dotnet/interactive/pull/344/files#diff-0202f42a93b1b0cde05b190029606ea9R11-R28